pinot-query-planner/src/main/java/org/apache/pinot/query/routing/WorkerManager.java

```diff
@@ -228,8 +234,13 @@ private void assignWorkersToPartitionedLeafFragment(DispatchablePlanMetadata met
           tableName, i);
       ServerInstance serverInstance =
           pickEnabledServer(partitionInfo._fullyReplicatedServers, enabledServerInstanceMap, indexToPick++);
-      Preconditions.checkState(serverInstance != null,
-          "Failed to find enabled fully replicated server for table: %s, partition: %s in table: %s", tableName, i);
+      if (serverInstance == null) {
```

Comment:
are we able to catch this exception all the way at top? If so, we don't have to add the these many `throws ProcessingException`